src/features/modules.md

```diff
@@ -126,14 +126,9 @@ Another difference relates to the `async` attribute, which causes the script to
 
 You may have noticed we’re using the `.mjs` file extension for modules. On the Web, the file extension doesn’t really matter, as long as the file is served with [the JavaScript MIME type `text/javascript`](https://html.spec.whatwg.org/multipage/scripting.html#scriptingLanguages:javascript-mime-type). The browser knows it’s a module because of the `type` attribute on the script element.
 
-Still, we recommend using the `.mjs` extension for modules, for two reasons:
+During development, the `.mjs` file extension makes it crystal clear that a file is a module as opposed to a regular script. (It’s not always possible to tell just by looking at the code.) As mentioned before, modules are treated differently than regular scripts, so the difference is hugely important! Additionally, you may want to configure your editor to treat `.mjs` files as `.js` files to get syntax highlighting. Most modern editors already do this by default.
 
-1. During development, it makes it crystal clear that the file is a module as opposed to a regular script. (It’s not always possible to tell just by looking at the code.) As mentioned before, modules are treated differently than regular scripts, so the difference is hugely important!
-1. It’s consistent with Node.js, where [the experimental modules implementation](https://nodejs.org/api/esm.html) only supports files with the `.mjs` extension by default.
```

Comment:
about the first point, in commonjs era nobody ever had an issue in writing modules as `.js` script, so that sentence is overly opinionated/forviant for no reason

the last point is not consistent, as NodeJS is currently already capable of loading `.js` files as ESM

Reply:
> about the first point, in commonjs era nobody ever had an issue in writing modules as .js script, so that sentence is overly opinionated/forviant for no reason

Please stay on topic. CommonJS "modules" are not ECMAScript modules, as in JavaScript programs that are parsed using the Module Goal. The article describes how modules are treated differently from classic scripts, and this section refers to that. 

> the last point is not consistent, as NodeJS is currently already capable of loading .js files as ESM https://nodejs.org/api/esm.html#esm_enabling

It is with additional configuration, but not by default, as our current article states.

Reply:
additional configuration does not need `.mjs` so that sentence should at least be modified:

> It’s consistent with Node.js, where the experimental modules implementation only supports, _unless configured differently_, files with the `.mjs` extension by default

the current sentence bases an argument over an _experimental_ flag, which should never be encouraged in references, and sounds like the only option, easily misleading readers that `.mjs` is consistent with NodeJS, which won't likely be the common case once the flag disappears.

Reply:
That's what "by default" means. You have a file, and you want to run it as a module in Node.js. If it's `.mjs`, it just works. If not, you need additional configuration.

Reply:
if you have `{"type": "module"}` in `/package.json` the default is another, same if you received your laptop configured by somebody responsible for laptops in the company.

I understand that sentence and all I am saying is that, as is, it's easily misleading and underlying that `.mjs` might be not necessary in some case would improve it.

If not, it feels like this whole document is too focused on telling people to use `.mjs`, which should be a developer choice, not something sold as subjective best/practice.

Reply:
No one is claiming that it is necessary in all cases, though -- in fact, the article already includes a counter-example (it explicitly states that on the web, extensions don't matter). Once again, you're putting up a strawman argument that I'm not interested in getting into. Please stop.